core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -209,6 +211,41 @@ private void enrichTopicInfoList(
     }
   }
 
+  private void setHasOpenRequestBooleans(
+      TopicOverviewInfo topicInfo, String topicName, String envId, int tenantId) {
+    topicInfo.setHasOpenACLRequest(isACLRequestOpen(topicName, envId, tenantId));
+    topicInfo.setHasOpenTopicRequest(isTopicRequestOpen(topicName, envId, tenantId));
+    topicInfo.setHasOpenRequest(
+        topicInfo.isHasOpenACLRequest()
+            || isSchemaRequestOpen(topicName, envId, tenantId)
+            || topicInfo.isHasOpenTopicRequest());
+  }
+
+  private void setHasSchema(
+      TopicOverviewInfo topicInfo, String topicName, String envId, int tenantId) {
+    Map<String, Env> allEnv = manageDatabase.getEnvMap(tenantId);
+    if (!allEnv.isEmpty()) {
+      Env topicEnv = allEnv.get(envId);
+      if (topicEnv.getAssociatedEnv() != null && topicEnv.getAssociatedEnv().getId() != null) {
+        List<MessageSchema> schema =
```

Comment:
instead of retrieving schema, can we get any count to optimize this

Reply:
Yeah we can definitely do counts.